request/configure.go

```diff
@@ -0,0 +1,141 @@
+package request
+
+import (
+	goerrors "errors"
+	"io/ioutil"
+	"os"
+	"os/user"
+	"path/filepath"
+	"strings"
+
+	"github.com/browserpass/browserpass-native/errors"
+	"github.com/browserpass/browserpass-native/response"
+	log "github.com/sirupsen/logrus"
+)
+
+func configure(request request) {
+	responseData := response.MakeConfigureResponse()
+
+	// Check that each and every store in the settings exists and is accessible.
+	// Then read the default configuration for these stores (if available).
+	for _, store := range request.Settings.Stores {
+		normalizedStorePath, err := normalizePasswordStorePath(store.Path)
+		if err != nil {
+			log.Errorf(
+				"Inaccessible path '%v' of the user-configured password store '%v': %+v",
+				store.Path, store.Name, err)
+			response.SendError(
+				errors.CodeInaccessiblePasswordStore,
+				"The password store is not accessible",
+				&map[string]string{"error": err.Error(), "name": store.Name, "path": store.Path})
+			errors.ExitWithCode(errors.CodeInaccessiblePasswordStore)
+		}
+
+		store.Path = normalizedStorePath
+
+		responseData.StoreSettings[store.Name], err = readDefaultSettings(store.Path)
+		if err != nil {
+			log.Errorf(
+				"Unable to read the default settings of the user-configured password store '%v' in '%v': %+v",
+				store.Name, store.Path, err)
+			response.SendError(
+				errors.CodeUnreadablePasswordStoreDefaultSettings,
+				"Unable to read the default settings of the password store",
+				&map[string]string{"error": err.Error(), "name": store.Name, "path": store.Path})
+			errors.ExitWithCode(errors.CodeUnreadablePasswordStoreDefaultSettings)
+		}
+	}
+
+	// Check whether a store in the default location exists and is accessible.
+	// If there is at least one store in the settings, it is expected that there might be no store in the default location => do not return errors.
```

Comment:
Note: in this case the app returns `{"defaultStore": {"path": "", "settings": ""}}`. I figured it will be better than returning `{"defaultStore": null}`, but if you prefer the latter (or something else), I can adapt. However, this only happens if user provided at least one custom store in settings, otherwise instead of the above an error will be returned (read the comment on the line below).

Reply:
`null` vs `""` doesn't really matter from my end; if the user is providing a store, then the default store is ignored anyway :-). So what you've done is just fine.